sklearn/preprocessing/tests/test_data.py

```diff
@@ -701,6 +703,62 @@ def test_scaler_without_centering():
     assert_array_almost_equal(X_csc_scaled_back.toarray(), X)
 
 
+def _check_attributes_scalers(scaler_1, scaler_2):
+    assert scaler_1.mean_ == scaler_2.mean_
+    assert scaler_1.var_ == scaler_2.var_
+    assert scaler_1.scale_ == scaler_2.scale_
+    assert scaler_1.n_samples_seen_ == scaler_2.n_samples_seen_
+
+
+def test_scaler_return_identity():
+    # test that the scaler return identity when with_mean and with_std are
+    # False
+    X_dense = np.array([[0, 1, 3],
+                        [5, 6, 0],
+                        [8, 0, 10]])
+    X_csr = sparse.csr_matrix(X_dense)
+    X_csc = X_csr.tocsc()
+
+    transformer_dense = StandardScaler(with_mean=False, with_std=False)
+    X_trans_dense = transformer_dense.fit_transform(X_dense)
+
+    transformer_csr = clone(transformer_dense)
+    X_trans_csr = transformer_csr.fit_transform(X_csr)
+
+    transformer_csc = clone(transformer_dense)
+    X_trans_csc = transformer_csc.fit_transform(X_csc)
+
+    assert_array_almost_equal(X_trans_csr.A, X_csr.A)
+    assert_array_almost_equal(X_trans_csc.A, X_csc.A)
```

Comment:
What is this `.A` attributed on sparse matrices? It's not documented in the docstring. I would rather use the more explicit: `X_trans_csr.toarray(), X_csr.toarray()` instead.

Also, didn't we decide to favor `assert_allclose` instead of `assert_array_almost_equal`?

Also here X_dense has an integer dtype. I think We should rather make this test on `np.float32` or `np.float64`.

If we test on integers, we should in addition check that we get the expected dtype. For standard scaling I would expect to always get floating point values on the output, even with `with_mean=False, with_std=False` to keep consistency. But if you disagree, I can probably be convinced otherwise.

Reply:
> If we test on integers, we should in addition check that we get the expected dtype. For standard scaling I would expect to always get floating point values on the output, even with with_mean=False, with_std=False to keep consistency. But if you disagree, I can probably be convinced otherwise.

This is a good point to keep in mind. However, I would delegate this part to the issues/PRs which try to preserve the dtype: https://github.com/scikit-learn/scikit-learn/issues/11000